pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/StatusAcknowledgeServlet.java

```diff
@@ -0,0 +1,154 @@
+            response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, e.getMessage());
+        }
+    }
+
+    /**
+     *  Checks whether all the fields are present
+     * @param acknowledgement the acknowledement object containing request data
+     * @return true if not all fields have data else return false
+     */
+    private boolean isObjectNullOrEmpty(Acknowledgement acknowledgement) {
+        return null == acknowledgement ||
+                Stream.of(acknowledgement.getId(), acknowledgement.getMessage(),
+                        acknowledgement.getSender(),acknowledgement.getStatus())
+                        .anyMatch(Objects::isNull);
+    }
+
+    /**
+     * Process the data in acknowldegment and create node if the module
+     * for which the acknowledgement has data contains supported locale
+     * @param acknowledgement request data
+     * @param module module corresponding to the UUID in the request data
+     * @param moduleLocale list of locales in the module
+     * @throws PersistenceException signals that request data could not be saved
+     */
+    private void processAcknowledgementRequest(Acknowledgement acknowledgement, Module module,
+                                               List<Resource> moduleLocale) throws PersistenceException {
+
+        for (Resource locale : moduleLocale) {
+            //defensive programming: double check that only for en_US locale the status node is created
+            if(locale.getName().equalsIgnoreCase("en_US")) {
+                createStatusNode(locale, module, acknowledgement);
+                break;
+            }
+        }
+    }
+
+    private boolean hasLocale(List<Resource> moduleLocale, String locale) {
+        return moduleLocale.stream().anyMatch(ml -> ml.getName().equalsIgnoreCase(locale));
+    }
+
+    private Acknowledgement getAcknowledgementData(@NotNull SlingHttpServletRequest request) throws IOException {
+        TransformToPojo transformToPojo = new TransformToPojo();
+        return transformToPojo.fromJson(Acknowledgement.class, request.getReader());
+    }
+
+    private void createStatusNode(Resource moduleLocale, Module module, Acknowledgement acknowledgement) throws PersistenceException {
+        Locale locale = LocaleUtils.toLocale(moduleLocale.getName());
+        Status status = module.getReleasedVersion(locale).get().status().getOrCreate();
+        status.status().set(acknowledgement.getStatus());
+        status.message().set(acknowledgement.getMessage());
+        status.sender().set(acknowledgement.getSender());
+        status.getResourceResolver().commit();
+    }
+
+    /**
+     * Retrieves the module corresponding to the uuid
+     * @param uuid id of the module
+     * @param request sling request
+     * @return resource correponding to the uuid
+     * @throws RepositoryException
+     */
+    private Resource getResourceByUuid(String uuid, SlingHttpServletRequest request) throws RepositoryException {
```

Comment:
I wonder if this method (which is generic enough to be useful in other places) would be better suited to live in the `JcrQueryHelper` class.

Reply:
Yes. Since GetObjectByUUID also makes use of this, I was going to raise a JIRA so that GetObjectByUUID can also be refactored.

Reply:
I have raised https://projects.engineering.redhat.com/browse/CCS-3537 for refactoring.